pkg/commands/config.go

```diff
@@ -39,6 +41,16 @@ var (
 	baseImageOverrides map[string]name.Reference
 )
 
+// SetDefaultBaseImage enables programmatically overriding the base image
+func SetDefaultBaseImage(baseImage string) error {
```

Comment:
I don't love this being a package global. Can we add this to the BuildOptions and and plumb it down through to https://github.com/google/ko/blob/1b023d30207f385020a5829deec7c147719a1ff9/pkg/commands/resolver.go#L82?

It's a private function, so we can just add another parameter to `getBaseImage` and pass in the `defaultBaseImage`

Reply:
Good point. I removed the exported function and added `BaseDir` to `BuildOptions`.

I also found that we could set the variable without modifying the `getBaseImage` function signature, but please let me know if you'd rather add a parameter.

On a related note, I was thinking to make `UserAgent` an overrideable option, but it's used by both Builder (to fetch the base image) and by Publisher (to push to the registry). We could have two different user-agent options, or introduce some shared build/publish option.

Reply:
> please let me know if you'd rather add a parameter

I think I'd prefer as a parameter, because this is still mutating package-global state. If you call `MakeBuilder` twice from the same program, they would interfere with each other.

On that note, we might want to rename `MakeBuilder` to `NewBuilder`? I think that's more in line with idiomatic constructor names in go?

> We could have two different user-agent options, or introduce some shared build/publish option.

I'm fine with whatever's easiest -- if I were implementing this I'd probably just add a second publisher option, but I'm curious about your opinion as a consumer. I know it's kind of gross to have to specify that twice.

Reply:
Great point, I've changed it to a parameter.

I've also added `UserAgent` fields to both `BuildOptions` and `PublishOptions`. Having to specify them both programmatically didn't seem like a big deal.